File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;
import java.io.*;

class Solution {
    int visited[];
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new int[n];
        
        for (int i=0; i < n; i++) {
            if (visited[i] == 0) {
                answer += 1;
                DFS(i, computers, n);
            }
        }
        return answer;
    }
    
    public void DFS(int i, int[][] computers, int n) {
        visited[i] = 1;
        
        for (int j=0; j < n; j++) {
            if (visited[j] == 0 && computers[i][j] == 1){
                DFS(j, computers, n);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
## DFS 풀이
def dfs(n, computers, com, visited) :
    visited[com] = 1
    for connect in range(n) :
        ## 네트워크가 연결된 컴퓨터인 경우
        if connect != com and computers[com][connect] == 1 :
            if visited[connect] == 0 :
                dfs(n, computers, connect, visited)

def solution(n, computers):
    answer = 0
    visited = [0 for j in range(n)]
    
    for com in range(n) :
        if visited[com] == 0 :
            dfs(n, computers, com, visited)
            answer += 1
    return answer